apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c

```diff
@@ -56,33 +55,326 @@ struct arastorage_data_type_s {
+* @precondition         none
+* @postcondition        none
+*/
+static void db_cleanup(const char *rel)
+{
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "REMOVE RELATION %s;", rel);
+	db_exec(g_query);
+}
+
+/**
+* @testcase             db_insert
+* @brief                To insert data into db
+* @scenario             insert data into db
+* @apicovered           db_exec
+* @precondition         none
+* @postcondition        none
+*/
+static void db_insert(const char *rel_name)
+{
+	g_flag = true;
+	db_result_t ret;
+
+	ret = db_init();
+	if (ret != DB_OK) {
+		g_flag = false;
+		printf("\ndb_init FAIL\n");
+		return;
+	}
+
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE RELATION %s;", rel_name);
+	ret = db_exec(g_query);
+	if (ret != DB_OK) {
+		g_flag = false;
+		printf("\ndb_exec FAIL\n");
+		return;
+	}
+
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE ATTRIBUTE %s DOMAIN int IN %s;", g_attribute_set[0], rel_name);
+	ret = db_exec(g_query);
+	if (ret != DB_OK) {
+		g_flag = false;
+		printf("\ndb_exec FAIL\n");
+		db_cleanup(rel_name);
+		db_deinit();
+		return;
+	}
+
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE ATTRIBUTE %s DOMAIN long IN %s;", g_attribute_set[1], rel_name);
+	ret = db_exec(g_query);
+	if (ret != DB_OK) {
+		g_flag = false;
+		printf("\ndb_exec FAIL\n");
+		db_cleanup(rel_name);
+		db_deinit();
+		return;
+	}
+
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE ATTRIBUTE %s DOMAIN string(32) IN %s;", g_attribute_set[2], rel_name);
+	ret = db_exec(g_query);
+	if (ret != DB_OK) {
+		g_flag = false;
+		printf("\ndb_exec FAIL\n");
+		db_cleanup(rel_name);
+		db_deinit();
+		return;
+	}
+
+	memset(g_query, 0, QUERY_LENGTH);
+	sprintf(g_query, "CREATE INDEX %s.%s TYPE %s;", rel_name, g_attribute_set[0], RELATION_INDEX);
+	ret = db_exec(g_query);
+	if (ret != DB_OK) {
+		g_flag = false;
+		printf("\ndb_exec FAIL\n");
+		db_cleanup(rel_name);
+		db_deinit();
+		return;
+	}
+
+	int index;
+	for (index = 0; index < DATA_SET_NUM; index++) {
+		memset(g_query, 0, QUERY_LENGTH);
+		snprintf(g_query, QUERY_LENGTH, "INSERT (%d, %ld, \'%s\') INTO %s;", g_arastorage_data_set[index].int_value,
+			g_arastorage_data_set[index].long_value, g_arastorage_data_set[index].string_value,
+			rel_name);
+		ret = db_exec(g_query);
+		if (ret != DB_OK) {
+			g_flag = false;
+			printf("\ndb_exec FAIL\n");
+			db_cleanup(rel_name);
+			db_deinit();
+			return;
+		}
+	}
+
+	g_cursor = NULL;
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "SELECT %s, %s, %s FROM %s WHERE %s > 2;", g_attribute_set[0], g_attribute_set[1], g_attribute_set[2], rel_name, g_attribute_set[0]);
+	g_cursor = db_query(g_query);
+	if (g_cursor == NULL) {
+		g_flag = false;
+		printf("\ndb_query FAIL\n");
+		db_cleanup(rel_name);
+		db_deinit();
+		return;
+	}
+
+	ret = db_deinit();
+	if (ret != DB_OK) {
```

Comment:
this code also duplicated.
you'd better implement goto_error_exec & goto_error_query